test/ecto/integration/aggregate_function_type_test.exs

```diff
@@ -0,0 +1,193 @@
+        types: [uid: "Int16", updated: "DateTime", name: "String"]
+      )
+
+      assert "agg_test_users"
+             |> select([u], %{
+               uid: u.uid,
+               updated: max(u.updated),
+               name: fragment("argMaxMerge(?)", u.name)
+             })
+             |> group_by([u], u.uid)
+             |> TestRepo.all() == [
+               %{
+                 uid: 1231,
+                 updated: ~N[2020-01-02 00:00:00],
+                 name: "Jane"
+               }
+             ]
+    end
+
+    test "materialized view and null engine" do
+      TestRepo.query!("""
+      CREATE TABLE agg_test_users (
+        uid Int16,
+        updated SimpleAggregateFunction(max, DateTime),
+        name AggregateFunction(argMax, String, DateTime)
+      ) ENGINE AggregatingMergeTree ORDER BY uid
+      """)
+
+      on_exit(fn -> TestRepo.query!("DROP TABLE agg_test_users") end)
+
+      TestRepo.query!("""
+      CREATE TABLE agg_test_users_null (
+        uid Int16,
+        updated DateTime,
+        name String
+      ) ENGINE Null
+      """)
+
+      on_exit(fn -> TestRepo.query!("DROP TABLE agg_test_users_null") end)
+
+      TestRepo.query!("""
+      CREATE MATERIALIZED VIEW agg_test_users_mv TO agg_test_users AS
+        SELECT uid, updated, arrayReduce('argMaxState', [name], [updated]) name
+        FROM agg_test_users_null
+      """)
+
+      on_exit(fn -> TestRepo.query!("DROP MATERIALIZED VIEW agg_test_users_mv") end)
+
+      # assert {4, _} =
+      #          TestRepo.insert_all(
+      #            "agg_test_users_null",
+      #            [
+      #              [uid: 1231, updated: ~N[2020-01-02 00:00:00], name_stub: "Jane"],
+      #              [uid: 1231, updated: ~N[2020-01-01 00:00:00], name_stub: "John"]
+      #            ],
+      #            types: [
+      #              uid: "Int16",
+      #              updated: "DateTime",
+      #              name_stub: "String"
+      #            ]
+      #          )
+
+      TestRepo.query!(
+        "INSERT INTO agg_test_users_null FORMAT RowBinaryWithNamesAndTypes",
+        _rows = [
+          [1231, ~N[2020-01-02 00:00:00], "Jane"],
+          [1231, ~N[2020-01-01 00:00:00], "John"]
+        ],
+        names: ["uid", "updated", "name_stub"],
+        types: ["Int16", "DateTime", "String"]
+      )
+
+      assert "agg_test_users"
+             |> select([u], %{
+               uid: u.uid,
+               updated: max(u.updated),
+               name: fragment("argMaxMerge(?)", u.name)
+             })
+             |> group_by([u], u.uid)
+             |> TestRepo.all() == [
+               %{
+                 uid: 1231,
+                 updated: ~N[2020-01-02 00:00:00],
+                 name: "Jane"
+               }
+             ]
+    end
+  end
+
+  # TODO move to Ch.input/1
+  # https://clickhouse.com/docs/en/sql-reference/table-functions/input
+  @spec input(Keyword.t() | module) :: Ecto.Query.t()
```

Comment:
- [ ] return `Ecto.Queryable.t` instead of raw query, like `%Ch.Input{structure: ...}`, that would provide more flexibility in the adapter.